handler/delete.go

```diff
@@ -21,33 +21,32 @@ import (
 
 	"github.com/go-kit/kit/log"
 	"github.com/go-kit/kit/log/level"
-	"github.com/julienschmidt/httprouter"
 	"github.com/prometheus/client_golang/prometheus"
 	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/prometheus/common/route"
 
 	"github.com/prometheus/pushgateway/storage"
 )
 
 // Delete returns a handler that accepts delete requests.
 //
 // The returned handler is already instrumented for Prometheus.
-func Delete(ms storage.MetricStore, jobBase64Encoded bool, logger log.Logger) func(http.ResponseWriter, *http.Request, httprouter.Params) {
-	var ps httprouter.Params
+func Delete(ms storage.MetricStore, jobBase64Encoded bool, logger log.Logger) func(http.ResponseWriter, *http.Request) {
 	var mtx sync.Mutex // Protects ps.
 
 	instrumentedHandler := promhttp.InstrumentHandlerCounter(
-		httpCnt.MustCurryWith(prometheus.Labels{"handler": "delete"}),
-		http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
-			job := ps.ByName("job")
+		HTTPCnt.MustCurryWith(prometheus.Labels{"handler": "delete"}),
```

Comment:
This can be removed by using similar wrap function used in api/v1/api.go

Reply:
Indeed. And if we have a helper function for that here in handler, we can also use it in the `wrap` for the API and don't need to export HTTPCnt. Just a minute…